spark_films/best_films_task3.py

```diff
@@ -0,0 +1,147 @@
+from pyspark.sql import SparkSession
+from pyspark import SparkConf
+import pyspark.sql.types as t
+import pyspark.sql.functions as f
+from pyspark.sql.window import Window
+
+
+spark = SparkSession.builder\
+    .master("local")\
+    .appName("Practise")\
+    .config(conf=SparkConf())\
+    .getOrCreate()
+
+path = "/home/nikolay/Projects/pyspark_learn/basic.tsv"
+df_spark = spark.read.csv(path, sep="\t", header=True)
+
+
+codes_path = "/home/nikolay/Projects/pyspark_learn/ranking.tsv"
+ranking_df = spark.read.csv(codes_path, sep='\t', header=True)
+
+code_scheme = t.StructType([
+    t.StructField("id", t.StringType(), True),
+    t.StructField("ranking", t.DoubleType(), True),
+    t.StructField("numVotes", t.IntegerType(), True)
+])
+
+best_films = df_spark.join(ranking_df, on="tconst", how="left")
+
+#  divide column with genres into separate words
+split_genre = f.explode(f.split(f.col("genres"), ","))
+
+window = Window.partitionBy("genres").orderBy(f.col("averageRating"))
+best_films = best_films.withColumn("yearRange", f.lit("2020-today"))
+
+#  best films sorted by genre and ranked by rating for the period from 2020 to today
```

Comment:
Commented code shouldn't be present in final version of the PR

Reply:
Changes)